File: Mntone.SplatoonClient/Mntone.SplatoonClient.Entities/Stage.cs
using Mntone.SplatoonClient.Entities.Internal;
using System;
using System.Runtime.Serialization;

namespace Mntone.SplatoonClient.Entities
{
	/// <summary>
	/// Stage information
	/// </summary>
	[DataContract]
	[System.Diagnostics.DebuggerDisplay("Name = {this.Name}")]
	public sealed class Stage
	{
		private Stage() { }

		internal Stage(string name, Uri imageUri)
		{
			this.Name = name;
			this.ImageUri = imageUri;
		}

		/// <summary>
		/// Stage name
		/// </summary>
		[DataMember(Name = "name", IsRequired = true)]
		public string Name { get; private set; }

		/// <summary>
		/// Stage image uri
		/// </summary>
		public Uri ImageUri { get; private set; }

		[DataMember(Name = "asset_path", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string ImageUriImpl
		{
			get { return this.ImageUri.PathAndQuery; }
			set { this.ImageUri = value.ToAbsoluteUriFromPathAndQuery(); }
		}
	}
}

File: net.mntone.splatoonclient/net.mntone.splatoonclient.entities/src/main/java/net/mntone/splatoonclient/entities/Stage.java
package net.mntone.splatoonclient.entities;

import org.json.JSONObject;

import java.net.MalformedURLException;
import java.net.URL;

public final class Stage
{
	private final String _name;
	private final URL _imageUri;

	Stage(final String name, final URL imageUri)
	{
		this._name = name;
		this._imageUri = imageUri;
	}

	Stage(final JSONObject json) throws MalformedURLException
	{
		this._name = json.getString("name");
		this._imageUri = new URL(json.getString("asset_path"));
	}

	public String getName()
	{
		return this._name;
	}

	public URL getImageUri()
	{
		return this._imageUri;
	}
}